### PR TITLE
added entity `Category`

### PR DESCRIPTION
**Pull request by StoneBlood-bit**

### CODE REVIEW

File: src/main/java/mate/academy/controller/CategoryController.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Category manager", description = "Endpoints for maneging book`s categories")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/categories")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/controller/CategoryController.java]:
```suggestion
@RequestMapping("/categories")
```

File: src/main/java/mate/academy/dto/book/CreateBookRequestDto.java [hunk 2]
```diff
@@ -19,4 +20,5 @@ public class CreateBookRequestDto {
     private BigDecimal price;
     private String description;
     private String coverImage;
+    private List<Long> categoryIds;
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/dto/book/CreateBookRequestDto.java:25]: let's add validation

File: src/main/java/mate/academy/model/Category.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Data
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/model/Category.java]: Don't use `@Data` for entities, find articles to read about problems

File: src/main/java/mate/academy/service/category/CategoryServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    public void deleteById(Long id) {
+        categoryRepository.deleteById(id);
+    }
+
+    @Override
+    public List<BookDtoWithoutCategoryIds> getBooksByCategoryId(Long id) {
+        List<Book> bookList = bookRepository.findByCategories_Id(id);
+        return bookList.stream()
+                .map(bookMapper::toDtoWithoutCategories)
+                .collect(Collectors.toList());
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/service/category/CategoryServiceImpl.java]:
```suggestion
                .toList();
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/mate/academy/repository/BookRepository.java [hunk 5]
```diff
@@ -1,7 +1,10 @@
 package mate.academy.repository;
 
+import java.util.List;
 import mate.academy.model.Book;
 import org.springframework.data.jpa.repository.JpaRepository;
 
 public interface BookRepository extends JpaRepository<Book, Long> {
+    List<Book> findByCategories_Id(Long categoryId);
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/repository/BookRepository.java]:
```suggestion
    List<Book> findByCategoriesId(Long categoryId);
```

File: src/main/java/mate/academy/service/category/CategoryService.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+    CategoryDto getById(Long id);
+
+    CategoryDto save(CategoryDto categoryDto);
+
+    CategoryDto update(Long id, CategoryDto categoryDto);
+
+    void deleteById(Long id);
+
+    List<BookDtoWithoutCategoryIds> getBooksByCategoryId(Long id);
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/service/category/CategoryService.java]: better move it to BookService

**Review by liliia-ponomarenko — APPROVED** [review]
Good job! A few minor comments but please fix them before merging ;)

**Pull request merged**